cloudify_cli/tests/commands/test_sites.py

```diff
@@ -0,0 +1,96 @@
+########
+# Copyright (c) 2013-2019 Cloudify Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+from .test_base import CliCommandTest
+from cloudify_cli.exceptions import CloudifyValidationError, CloudifyCliError
+
+
+class SitesTest(CliCommandTest):
+    system_exit = {
+        'err_str_segment': '2',
+        'exception': SystemExit
+    }
+
+    def setUp(self):
+        super(SitesTest, self).setUp()
+        self.use_manager()
+
+    def test_get_missing_name(self):
+        outcome = self.invoke('cfy sites get', **self.system_exit)
+        self.assertIn('Missing argument "name"', outcome.output)
+
+    def test_get_invalid_name(self):
+        self.invoke(
+            "cfy sites get ' ' ",
+            err_str_segment='ERROR: The `name` argument contains illegal '
+                            'characters',
+            exception=CloudifyValidationError
+        )
+
+        self.invoke(
+            "cfy sites get :bla",
+            err_str_segment='ERROR: The `name` argument contains illegal '
```

Comment:
is the illegal characters something general in the rest fields or something specific? but I think maybe add a paramtized decorator of pytest and loop over all the options

Reply:
It is something general, only letters, digits and the characters "-", "." and "_" are allowed.

Reply:
Oh I remember